src/python/pants/option/parser.py

```diff
@@ -576,9 +578,15 @@ def expand(val_or_str):
                 if val_or_str.startswith("@@"):  # Support a literal @ for fromfile values via @@.
                     return val_or_str[1:]
                 else:
-                    fromfile = val_or_str[1:]
+                    if val_or_str.startswith("@?"):  # Support an optional fromfile value via @?.
+                        fromfile, optional = val_or_str[2:], True
+                    else:
+                        fromfile, optional = val_or_str[1:], False
+                    fromfile_path = Path(get_buildroot(), fromfile)
                     try:
-                        contents = Path(get_buildroot(), fromfile).read_text()
+                        if optional and not fromfile_path.exists():
+                            return None
+                        contents = fromfile_path.read_text()
                         if fromfile.endswith(".json"):
```

Comment:
It may be nice to update these to use `Path.suffix` but I was avoiding making unnecessary changes in this PR.